### ISSUE TITLE
Framework: Resolve React 15.2 unknown props warnings

### ISSUE THREAD

**Issue by aduth** [Automattic/wp-calypso#7413]
As of React 15.2, when attempting to render a DOM element with an unknown prop, a warning will be logged in development environments. For example:

```text
warning.js:44 Warning: Unknown prop `compact` on <div> tag. Remove this prop from the element. For details, see https://fb.me/react-unknown-prop
```

In most cases, this is due to unrestrained application of a component's own props using the spread operator on a DOM element child. This can often be resolved by being more explicit in the props being passed to a DOM element. For cases where it's intentional to apply all additional props to a child except for a component's own custom props, using [`_.omit`](lodash.com/docs#omit) can be used to remove those custom props. Similarly, if intending to apply only a subset of a component's own props, [`_.pick`](lodash.com/docs#pick) can be used.

The following is an incomplete list of components which, as of this issue creation, emit warnings and need to be updated. Please add to this list with additional components, include your name next to a component to claim ownership for updating, and include links to relevant pull requests.

- [x] `<Button />` (@aduth #7104)
- [x] `<Dialog />` (@aduth #7408)
- [x] `<SiteSelector />` (@aduth #7323, #7478, #7622)
- [x] `<FormTextInput />` (@aduth #7411)
- [x] `<FormPasswordInput />` (@tyxla #8273)
- [x] `<Popover />` (@retrofox).
- [x] `<DropdownSelector />`  (@retrofox).
- [x] `<PostSchedule />` (@retrofox).
- [x] `<Layout />` (@roccotripaldi #7443)
- [x] `<FormSettingExplanation />` (@roccotripaldi #7496)
- [x] `<ExternalLink />` (@gwwar #7444)
- [x] `<Card />` (@ockham #7407, #7517)
- [x] `<InfiniteList />` (@gwwar #7533)
- [x] `<CountedTextarea />` (@copons #7577)
- [x] `<DatePicker />` (@copons #7569)
- [x] `<DisconnectJetpackButton />` (@tyxla #8131)
- [x] `<FormTextInputWithAffixes />` (@copons #7578)
- [x] `<FormSelect />` (@gziolo #7559)
- [x] `<FormsButton />` (@gziolo #7572)
- [x] `<FormCountrySelect />` (@gziolo #7559)
- [x] `<SelectOptGroups />` (@copons #7579)
- [x] `<ReaderSiteStreamLink />` (@bluefuton #7789)
- [x] `<CreditCardForm />` (@gziolo #7561)
- [x] `<DomainDetailsForm />` (@gziolo #7561)
- [x] `<Form />` (@gziolo #7561)
- [x] `<EditContactInfoFormCard />` (@gziolo #7561)
- [x] `<ProfileLink />` (@tyxla #7802)
- [x] `<PopoverMenuItem />` (@aduth #7611)
- [x] `<PeopleProfile />` and others in People section (@roccotripaldi #8629, #8824, #8795)
- [x] `<EditorAuthor />` (@aduth #8694, @roccotripaldi #8795)
- [x] `<PressThisLink />` (@tyxla #8271)
- [x] `<Shortcode />` (@tyxla #8277)
- [x] `<SharingButtonsPreviewAction />` (@tyxla #8779)

**Comment by ockham** [comment]
I've noticed that in some cases, our [`lib/react-pass-to-children`](https://github.com/Automattic/wp-calypso/tree/master/client/lib/react-pass-to-children) helper is used for props transfer (e.g by `PreferencesData`). We should probably modify that helper to either accept a `props` object (instead of a react element whose `props` are used), or to accept an omission list. I'm leaning towards the former.

**Comment by ockham** [comment]
Oops, just noticed #7323 which resolved the `PreferencesData` differently.

**Comment by aduth** [comment]
@retrofox : Are we able to check `<Popover />` off this list with the merging of #7217 ?

**Comment by retrofox** [comment]
@aduth I don't think so. Let me take a look but I'm afraid I didn't any change to get rid of these warnings.

**Comment by Copons** [comment]
It seems that `<DatePicker />` is not Calypso's fault but [react-day-picker](https://github.com/gpbl/react-day-picker)'s.

The warning is `Warning: Unknown props initialMonth, renderDay, localeUtils, onDayClick, onMonthChange, enableOutsideDays, onCaptionClick, captionElement on <div> tag. [...] in div (created by DayPicker)`
and look what happens in `DayPicker`:
https://github.com/gpbl/react-day-picker/blob/master/src/DayPicker.js#L19-L22
and then
https://github.com/gpbl/react-day-picker/blob/master/src/DayPicker.js#L476

tl;dr
`<DayPicker />` is an external component from `react-day-picker` that spreads its own props (that Calypso's `<DatePicker />` passes to it) into its `<div />`.

**Comment by aduth** [comment]
@Copons : I think it may be a matter of upgrading the dependency, since from [this line](https://github.com/gpbl/react-day-picker/blob/ba1f7d04782b755f88beb7b23c60a5d77dc25bb4/src/DayPicker.js#L464) it appears that the component's own props should be excluded from the spread. See also https://github.com/gpbl/react-day-picker/commit/26d3d816ae06f0f819d7366df5686625d4434e3e and https://github.com/gpbl/react-day-picker/pull/191 .

**Comment by blowery** [comment]
I _think_ ReaderSiteStream is fixed? Need to verify more, but it doesn't appear to be throwing to console and props and being omitted.

**Comment by tyxla** [comment]
@blowery `ReaderSiteStreamLink` has been fixed indeed (#7789) - I've verified that and updated the checklist.

**Comment by aduth** [comment]
I'm going to close this, as all tasks on the original list are now resolved. If others arise, they can be resolved on a component-by-component basis.
